[Obs AI] Add get_traces tool to retrieve distributed trace samples (#250952)
elastic/kibana: x-pack/solutions/observability/plugins/observability_agent_builder/server/tools/get_traces/README.md

# get_traces

Retrieves trace data (APM transactions/spans/errors) plus logs for one or more traces.

> This tool is KQL-driven: it finds one or more seed documents (logs or APM events) within the time range, extracts one or more `trace.id` values from those documents, then fetches APM events and logs for each trace.

Reviewer: This descriptions attempts to both describe the tool at a high level, and the implementation at the same time. Let's describe it at a high level only in the first sentence:

```suggestion
This tool finds traces: documents grouped  by `trace.id` within a given time range and for a given kql filter. 
```

If you want to elaborate on the implementation, you can add a "How it works" section below.

- Direct lookup by `traceId`
- Anchor-based lookup from logs (via `logId`, or `kqlFilter` + time range)

When anchoring from logs, the tool tries to extract the best correlation identifier from the anchor log (prefers `trace.id`). If the anchor only contains non-APM identifiers (e.g., `request.id`), APM results may be empty while logs still return.

## Examples

### Direct lookup by `traceId`

```jsonc
POST kbn://api/agent_builder/tools/_execute
{
  "tool_id": "observability.get_traces",
  "tool_params": {
    "start": "now-1h",
    "end": "now",
    "traceId": "abc123"
  }
}
```

### Anchor from a specific log entry (`logId`)

```jsonc
POST kbn://api/agent_builder/tools/_execute
{
  "tool_id": "observability.get_traces",
  "tool_params": {
    "logId": "abc123"
  }
}

> ```

Reviewer: Let's simplify the API and only have `kqlFIlter`. We can use the tool param description to help the LLM use this to search for specific logs or traces:

Retrieve specific trace:
```
kqlFilter: "trace.id: abc"
```

Retrieve trace for a particular log:
```
kqlFilter: "_id: abc"
```
